lib/date_to_str.c

```diff
@@ -54,6 +33,8 @@ date_to_str(size_t size, char buf[size], long date)
 		return;
 	}
 
-	(void) strftime(buf, size, "%Y-%m-%d", tm);
-	buf[size - 1] = '\0';
+	if (strftime(buf, size, "%Y-%m-%d", tm) == 0) {
+		(void) strtcpy(buf, "future", size);
```

Comment:
This looks fine - but of course the return isn't needed, and then the braces aren't needed.  Up to you whether you want to change that or want me to just merge it as is.

Reply:
Fixed.